src/overlays/actors/ovl_En_Weather_Tag/z_en_weather_tag.c

```diff
@@ -15,21 +15,34 @@ void EnWeatherTag_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnWeatherTag_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnWeatherTag_Update(Actor* thisx, GlobalContext* globalCtx);
 
-void func_80B31658(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31700(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31850(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31924(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B317A8(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31A10(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31B24(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31C20(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B316B0(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31758(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31800(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B318C0(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B319A0(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31AA0(EnWeatherTag* this, GlobalContext* globalCtx);
-void func_80B31CC0(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledCloudyHyruleMarket(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledCloudyHyruleMarket(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledCloudyLonLonRanch(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledCloudyLonLonRanch(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledCloudySnow(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledCloudySnow(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledRainLakeHylia(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledRainLakeHylia(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledCloudyDeathMountain(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledCloudyDeathMountain(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledCloudyRainThunderKakariko(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledCloudyRainThunderKakariko(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_SetSandstormIntensity(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_DisabledRainThunder(EnWeatherTag* this, GlobalContext* globalCtx);
+void EnWeatherTag_EnabledRainThunder(EnWeatherTag* this, GlobalContext* globalCtx);
+
+typedef enum {
+    EN_WEATHER_TAG_TYPE_0,
+    EN_WEATHER_TAG_TYPE_1,
+    EN_WEATHER_TAG_TYPE_2,
+    EN_WEATHER_TAG_TYPE_3,
+    EN_WEATHER_TAG_TYPE_4,
+    EN_WEATHER_TAG_TYPE_5,
+    EN_WEATHER_TAG_TYPE_6,
+    EN_WEATHER_TAG_TYPE_7
+} EnWeatherTagType;
```

Comment:
Since this enum is compared to the actor's params, I think it should be moved to the header.
Also, you could name each enum value based on the corresponding weather.

Reply:
agreed